lib/active_merchant/billing/gateways/omise.rb

```diff
@@ -194,8 +194,10 @@ def post_data(parameters)
 
       def https_request(method, endpoint, parameters=nil, options={})
         raw_response = response = nil
+        req_headers  = headers(options)
         begin
-          raw_response = ssl_request(method, url_for(endpoint), post_data(parameters), headers(options))
+          req_headers['Omise-Version'] = @api_version if @api_version
```

Comment:
The Omise api expects `Omise-Version` header only when user explicitly set this header.
So, when this doesn't get sent, Omise users expect the api version that has been set in their dashboard ( the setting in their account ).